Mathlib/RingTheory/HahnSeries/Basic.lean

```diff
@@ -289,6 +297,33 @@ theorem coeff_eq_zero_of_lt_orderTop {x : HahnSeries Γ R} {i : Γ} (hi : i < x.
   rw [orderTop_of_ne hx, WithTop.coe_lt_coe]
   exact Set.IsWF.not_lt_min _ _ hi
 
+/-- A leading coefficient of a Hahn series is the coefficient of a lowest-order nonzero term, or
+zero if the series vanishes. -/
+def leadingCoeff (x : HahnSeries Γ R) : R :=
+  if h : x = 0 then 0 else x.coeff (x.isWF_support.min (support_nonempty_iff.2 h))
+
+@[simp]
+theorem leadingCoeff_zero : leadingCoeff (0 : HahnSeries Γ R) = 0 :=
+  dif_pos rfl
+
+theorem leadingCoeff_of_ne {x : HahnSeries Γ R} (hx : x ≠ 0) :
+    x.leadingCoeff = x.coeff (x.isWF_support.min (support_nonempty_iff.2 hx)) :=
+  dif_neg hx
+
+theorem leadingCoeff_ne_iff {x : HahnSeries Γ R} : x.leadingCoeff ≠ 0 ↔ x ≠ 0 := by
```

Comment:
I might define `leadingCoeff_eq_iff` first and then prove this theorem as `leadingCoeff_eq_iff.not`